src/NServiceBus.Metrics/RawData/OccurrenceWriter.cs

```diff
@@ -0,0 +1,42 @@
+namespace NServiceBus.Metrics.RawData
+{
+    using System;
+    using System.IO;
+
+    static class OccurrenceWriter
+    {
+        const long Version = 1;
+
+        // WIRE FORMAT, Version: 1
+
+        //0                   1                   2                   3
+        //0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
```

Comment:
> What happens if we need to store floats or doubles?

Then we will come up with two more writers which are not currently needed.

> What assumptions needs the receiver of the wire format to take. 

My idea is to include the probe name + type in the message type header to enable proper decoding. The receiver will need to be aware of this mapping. We could embed the data part length in the header as well and make it self describing. Is this something, you'd think is better?

> Will the assumption be that values per data point are in a certain order

Yes. So far, every message contains time ordered entries, enabling easy merge-sort operation on the receiver side.

> So for example processingtimewireformat, criticaltimewireformat...?

```
 headers[Headers.EnclosedMessageTypes] = "NServiceBus.Metrics." + messageTypeName; 
```
where messageTypeName is: `Duration.ProcessingTime`, `Duration.CriticalTime`, `Signal.MessageCompleted` and so on.

Does it answer your questions @danielmarbach ?